### PR TITLE
Updated PHPCS: Packages and Debugger

### PR DESCRIPTION
**Pull request by kraftbj**
After #17406, updated the "requiredlist" files to match new PHPCS rules for packages and the debugger.

#### Changes proposed in this Pull Request:
* PHPCS

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
n/a

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

**Comment by kraftbj** [comment]
After 9.1 branches off and #17406 merges, we need to update base branch to `master`, then this can be merged.

**Comment by anomiex** [comment]
> After 9.1 branches off and #17406 merges, we need to update base branch to `master`, then this can be merged.

Once #17406 merges and its branch is deleted, Github will automatically set the base branch to master (or whatever we're calling it by then).

File: packages/constants/tests/php/test-constants.php [hunk 1]
```diff
@@ -156,7 +155,7 @@ public function test_jetpack_constants_get_constant_use_filter_value() {
 		$apply_filters_spy = new Spy(
 			'Automattic\Jetpack',
 			'apply_filters',
-			function ( $filter_name, $value, $name ) use ( $test_constant_value ) {
+			function ( $filter_name, $value, $name ) use ( $test_constant_value ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
Comment (anomiex) [inline packages/constants/tests/php/test-constants.php:158]: Could just remove the unused variables here, if you want.

I wouldn't recommend that for actual methods intended for being passed as callbacks, but this is just a closure.

File: packages/connection/tests/php/test_Error_Handler.php [hunk 2]
```diff
@@ -62,11 +62,11 @@ public function test_store_error() {
 
 		$stored_errors = $this->error_handler->get_stored_errors();
 
-		$this->assertEquals( 1, count( $stored_errors ) );
+		$this->assertSame( 1, count( $stored_errors ) );
```
Comment (anomiex) [inline packages/connection/tests/php/test_Error_Handler.php:65]: Random thought: Maybe we should have a sniff to prefer `assertCount` over things like this.

File: packages/sync/src/class-settings.php [hunk 3]
```diff
@@ -204,7 +204,7 @@ public static function update_settings( $new_settings ) {
 			}
 
 			// If we set the disabled option to true, clear the queues.
-			if ( ( 'disable' === $setting || 'network_disable' === $setting ) && ! ! $value ) {
+			if ( ( 'disable' === $setting || 'network_disable' === $setting ) && (bool) $value ) {
```
Comment (anomiex) [inline packages/sync/src/class-settings.php:207]: We could drop the `(bool)` entirely, PHP implicitly casts to bool in this situation anyway.

**Review by anomiex — APPROVED** [review]
Changes look sane to me, and tests are green. None of the comments I left are anything I'd hold up approval on.

**Review by anomiex — DISMISSED** [review]
Ugh, Github. It did change the branch to master, but still thinks it wants to apply all the patches from #17406 as part of this patch. Currently when this gets squash-and-merged those will wind up going away, but I have to wonder what would happen if some other patch touching those files got merged in the mean time.

**Review by anomiex — APPROVED** [review]

**Pull request merged**